### PR TITLE
PERS-527: Resolver usar todos os campos novos

### PR DESCRIPTION
**Pull request by LucasTornai**
#### What problem is this solving?

Today we don't have all fields necessary for the resolver in elasticsearch, so we need the unstable flag to get this information from the fq api. This PR solves some fields that require the fq, still lacking the category trees info.

#### How should this be manually tested?

Check the workspace https://tornai--footnotes.myvtex.com/

#### Checklist/Reminders

- [X] Updated `README.md`.
- [X] Updated `CHANGELOG.md`.
- [ ] Linked this PR to a Clubhouse story (if applicable).
- [ ] Updated/created tests (important for bug fixes).
- [ ] Deleted the workspace after merging this PR (if applicable).

#### Type of changes

✔️ | Type of Change
---|---
_ | Bug fix 
✔️ | New feature 
_ | Breaking change 
_ | Technical improvements

### CODE REVIEW

File: node/typings/Search.ts [hunk 1]
```diff
@@ -107,3 +99,93 @@ interface Breadcrumb {
   href: string
   name: string
 }
+
+interface BiggySearchProduct {
```
Comment (thalytafabrine) [inline node/typings/Search.ts:103]: can we rename these types without the `Biggy` prefix or would they conflict with types that already exist?

File: node/typings/Search.ts [hunk 2]
```diff
@@ -107,3 +99,93 @@ interface Breadcrumb {
...
+  link: string
+  oldPrice: number
+  price: number
+  brand: string
+  brandId: string
+  installment: BiggyInstallment
+  measurementUnit: string
+  unitMultiplier: number
+  tax: number
+
```
Comment (thalytafabrine) [inline node/typings/Search.ts]: there are some blank lines within the types, can we remove them?

File: node/typings/Search.ts [hunk 3]
```diff
@@ -107,3 +99,93 @@ interface Breadcrumb {
...
+  images: BiggyProductImages[]
```
Comment (thalytafabrine) [inline node/typings/Search.ts]:
```suggestion
  images: BiggyProductImage[]
```
considering that the type itself refers to a single image :)

File: node/commons/compatibility-layer.ts [hunk 4]
```diff
@@ -22,63 +17,168 @@ export const convertBiggyProduct = (
...
 
-const getSellersIndexedByXML = (product: any) => {
-  const { installment, price, oldPrice } = product
-  return [new VtexSeller('1', price, oldPrice, installment)]
+const getSellersIndexedByXML = (product: BiggySearchProduct): Seller[] => {
+  const price = product.price
+  const oldPrice = product.oldPrice
+  const installment = product.installment
+  const commertialOffer = buildCommertialOffer(price, oldPrice, installment, product.tax)
+  //return new VtexSeller(seller.id, price, oldPrice, installment, seller.tax)
```
Comment (thalytafabrine) [inline node/commons/compatibility-layer.ts]: 👀

File: node/typings/Search.ts (hunk 1 above)
Comment (LucasTornai) [reply node/typings/Search.ts:103]: Some of them would conflict. Even then, I like some sort of prefix (might be not biggy), but as to differ which DTO is the one that comes from the biggy DB, and the one that goes out.

File: node/commons/compatibility-layer.ts (hunk 4 above)
Comment (LucasTornai) [reply node/commons/compatibility-layer.ts]: 🙈

File: node/typings/Search.ts (hunk 3 above)
Comment (LucasTornai) [reply node/typings/Search.ts]: Done 👍

File: node/typings/Search.ts (hunk 2 above)
Comment (LucasTornai) [reply node/typings/Search.ts]: Done 👍

File: node/commons/compatibility-layer.ts [hunk 5]
```diff
@@ -24,71 +18,175 @@ export const convertBiggyProduct = (
...
-      product.extraInfo['marca'] ||
-      product.extraInfo['brand'] ||
-      '',
-    brandId: -1,
+    brand: product.brand || '',
+    brandId,
     link: product.url,
     description: product.description,
     items: skus,
-    sku: skus.find(sku => sku.sellers && sku.sellers.length > 0),
```
Comment (thalytafabrine) [inline node/commons/compatibility-layer.ts:55]: why did you remove this?

Comment (LucasTornai) [reply node/commons/compatibility-layer.ts:55]: The return in the compatibility layer should be the same return we have in the `fq` API, which is the type SearchProduct (with some other fields that are created on the run, mostly fields that are `categoryName: categoryValues`, this sku field does not exist in there (it's basically the items field)

**Review by thalytafabrine — APPROVED** [review]

File: node/commons/compatibility-layer.ts [hunk 6]
```diff
@@ -24,71 +18,178 @@ export const convertBiggyProduct = (
...
     convertSKU(product, indexingType, tradePolicy)
   )
 
+  const allSpecifications = product.productSpecifications.concat(getSKUSpecifications(product))
+
+  const specificationGroups = JSON.parse(product.specificationGroups)
+
+  const allSpecificationsGroups = [ ...Object.keys(specificationGroups) ]
+
+  const brandId = product.brandId ? Number(product.brandId) : -1
```
Comment (hiagolcm) [inline node/commons/compatibility-layer.ts:39]: Is it possible that the `brandId` is `0`? If it is, the `brandId` will become `-1` since the number `0` is falsy.

**Review by hiagolcm — APPROVED** [review]

File: node/commons/compatibility-layer.ts (hunk 6 above)
Comment (LucasTornai) [reply node/commons/compatibility-layer.ts:39]: As far as I'm aware, no. All stores I've checked that have only 1 brand start on brandId `1`, but mostly, they have a big number like `2000000` and `2000001` for instance.

File: node/commons/models/VtexSeller.ts [hunk 7]
```diff
@@ -26,7 +26,8 @@ class VtexSeller {
     sellerId: string,
     price: number,
     oldPrice: number,
-    installment: { value: number; count: number }
+    installment: { value: number; count: number },
+    tax?: number,
```
Comment (chrsmutti) [inline node/commons/models/VtexSeller.ts]: This class is no longer used, right? You changed it to use the already existing types for Seller and CommertialOffer,

Comment (LucasTornai) [reply node/commons/models/VtexSeller.ts]: Yeah, good point, it's not being used anywhere anymore. Imma delete it in the next commit.
